libcxx/include/string

```diff
@@ -1891,8 +1891,7 @@ private:
 #if !defined(_LIBCPP_HAS_NO_ASAN) && defined(_LIBCPP_INSTRUMENTED_WITH_ASAN)
     const void* __begin = data();
     const void* __end   = data() + capacity() + 1;
-    if (!__libcpp_is_constant_evaluated() && __begin != nullptr &&
```

Comment:
Why did the `__begin` != nullptr get removed?

Reply:
I should probably add it in the description of the PR, but `data()` cannot be nullptr, so I'm removing unnecessary check.